modules/sync/syncs/httpsynchronization.cpp

```diff
@@ -134,6 +158,82 @@ std::string HttpSynchronization::generateUid() {
     return fmt::format("{}/{}", _identifier, _version);
 }
 
+void HttpSynchronization::createSyncFile(bool fullySynchronized = true) const {
+    std::filesystem::path dir = directory();
+    std::filesystem::create_directories(dir);
+
+    dir.replace_extension("ossync");
+    std::ofstream syncFile(dir, std::ofstream::out);
+
+    syncFile << _ossyncVersionNumber << '\n' <<
+        (fullySynchronized ? _synchronizationToken : "Partial Synchronized") << '\n';
+
+    if (fullySynchronized)
+        return; //All files successfully downloaded, no need to write anything else to file.
+
+    //Store all files that successfully downloaded
+    for (std::string const& fileURL : _existingSyncedFiles) {
+        syncFile << fileURL << '\n';
+    }
+    //If we we fill _existingSyncedFiles before calling this func this loop is uneccessary
+    for (std::string const& fileURL : _newSyncedFiles) {
+        syncFile << fileURL << '\n';
+    }
+}
+
+bool HttpSynchronization::checkSyncFile() {
+    std::filesystem::path path = directory();
+    path.replace_extension("ossync");
+    //Check if file exists at all
+    if (!std::filesystem::is_regular_file(path)) {
+        return false;
+    }
+
+    //Read contents of file
+    std::ifstream file{ path };
```

Comment:
```suggestion
    std::ifstream file(path);
```

We don't use the newer initialization format for the cases where the older version also works